src/math/trig.c

```diff
@@ -94,9 +122,212 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
+	/* exponential */
+	if (sign) {
+		xn = -xn;
+		b_yn = -b_yn;
+	}
+
+	*cis.re = xn;
+	*cis.im = b_yn;
+	if ((type == EN_32B_CORDIC_CEXP) || (type == EN_16B_CORDIC_CEXP))
+		return cis;
 	/* Q2.30 format */
 	*th_cdc_fxp = th_rad_fxp;
 
 }
+/**
+ * Arguments	: int32_t sinvalue Q2.30
+ *		  int16_t numiters
+ *		  const int32_t LUT[30]
+ * Return Type	: int32_t
+ * Description	: Determines input value is scalar [logical 1 (true) logical 0 (false)]
```

Comment:
Is the description matching the next function code? I mean "if argument is a number, return one". The below code seems to do a lot more!

Reply:
Thanks. I will correct  and rephrase the description